Source/monster.cpp

```diff
@@ -1504,7 +1504,7 @@ void MonsterAttackPlayer(int i, int pnum, int hit, int minDam, int maxDam)
 			}
 		}
 	}
-	int dam = (minDam << 6) + GenerateRnd((maxDam - minDam + 1) << 6);
+	int dam = (minDam << 6) + (GenerateRnd(maxDam - minDam + 1) << 6);

```

Comment:
```suggestion
	int dam = (minDam + (GenerateRnd(maxDam - minDam + 1)) << 6;
```

Reply:
```suggestion
	int dam = (minDam << 6) + GenerateRnd(((maxDam - minDam) << 6) + 1);
```
is probably more in line with the original intent, this allows a monster to deal up to max damage without the extra fractional component